server.js

```diff
@@ -2,34 +2,102 @@ import express from "express";
 import cors from "cors";
 import mongoose from "mongoose";
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
 mongoose.Promise = Promise;
 
-// Defines the port the app will run on. Defaults to 8080, but can be overridden
-// when starting the server. Example command to overwrite PORT env variable value:
-// PORT=9000 npm start
+// Using a Mongoose model to model my data and to fetch data from the database.
+const Sale = mongoose.model("Sale", {
+  id: Number,
+  date: Number,
+  averagePrice: Number,
+  totalVolume: Number,
+  totalBagsSold: Number,
+  smallBagsSold: Number,
+  largeBagsSold: Number,
+  xLargeBagsSold: Number,
+  popularity: String
+});
+
+if(process.env.RESET_DB) {
+  const resetDataBase = async () => {
+    await Sale.deleteMany();
+    avocadoSalesData.forEach(singleSale => {
+      const newSale = new Sale(singleSale);
+      newSale.save();
+    })
+  }
+  resetDataBase();
+}
+
 const port = process.env.PORT || 8080;
-const app = express();
+const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors());
 app.use(express.json());
 
-// Start defining your routes here
+// --- Start of ROUTES ---
+
+// 1st route, the base path.
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send("Hello Avocado Sales Lover!");
 });
 
-// Start the server
+// 2nd route. To get a specific sale, using try-catch code block and the _id. The _id is from the MongoDB Compass.
+app.get("/sales/id/:id", async (req, res) => {
+  try {
+    const singleSale = await Sale.findById(req.params.id);
+    if (singleSale) {
+      res.status(200).json({
+        success: true,
+        body: singleSale
+      });
+    } else {
+      res.status(404).json({
+        success: false,
+        body: {
+          message: "Could not find the sale"
+        }
+      });
+    }
+  } catch(error) {
+    res.status(400).json({
+      success: false,
+      body: {
+        message: "Invalid id"
+      }
+    });
+  }
+});
+
+// 3rd route. To get an array of sales (if there are more than one) with specific parameters, using try-catch code block and query.
+app.get("/sales/", async (req, res) => {
+  const {region, averagePrice} = req.query;
+  const response = {
+    success: true,
+    body: {}
+  }
+  const matchAllRegex = new RegExp(".*");
+  const regionQuery = region ? region : matchAllRegex;
+  const averagePriceQuery = averagePrice ? averagePrice : matchAllRegex;
```

Comment:
Because averagePrice is a number, this regular expression will not work. Daniel had this problem during one of his lectures, and later posted another regex that should do the trick for datatype number:
{$gt: 0, $lt: 100}
So you can write:
`const averagePriceQuery = averagePrice ? averagePrice : {$gt: 0, $lt: 100};`